ambry-api/src/main/java/com/github/ambry/store/Store.java

```diff
@@ -40,6 +42,18 @@
    */
   StoreInfo get(List<? extends StoreKey> ids, EnumSet<StoreGetOptions> storeGetOptions) throws StoreException;
 
+  /**
+   * Returns the store info for the given ids asynchronously
+   * @param ids The list of ids whose messages need to be retrieved
+   * @param storeGetOptions A set of additional options that the store needs to use while getting the message
+   * @return a {@link CompletableFuture} that will eventually contain the {@link StoreInfo} for the given ids or the
+   *         {@link StoreException} if an error occurred
+   */
+  default CompletableFuture<StoreInfo> getAsync(List<? extends StoreKey> ids,
```

Comment:
I realized that all the async operations are for AmbryCloudRequests, we probably don't have to change the interface, we can just cast the Store to CloudStore and only implement these async operations in CloudStore.

Later if we want to do async operations in ambry-server, we can come back and change the interface.

Reply:
Sure. Updated to define the asynchronous methods only in CloubBlobStore and down cast the class when using it in AmbryCloudRequests.